src/storage/kv/rocksdb_engine.rs

```diff
@@ -260,11 +265,18 @@ impl Engine for RocksEngine {
         fail_point!("rockskv_async_snapshot", |_| Err(box_err!(
             "snapshot failed"
         )));
-        fail_point!("rockskv_async_snapshot_not_leader", |_| {
-            let mut header = ErrorHeader::default();
+        let not_leader = {
+            let mut header = kvproto::errorpb::Error::new();
             header.mut_not_leader().set_region_id(100);
-            Err(Error::Request(header))
+            header
+        };
+        let _not_leader = not_leader.clone();
```

Comment:
Is it necessary to declare a new variable?

Reply:
Yes, the `failpoints` macro moves `not_leader`.